### DISCUSSION TITLE
Wait Queue for Sessions - Open Mic/Stage - Pub Mode

### DISCUSSION THREAD

**Discussion by reinhardwh** [jamulussoftware/jamulus#2465] [Ideas]
Idea:
Offer a server configuration like:

10 Session Players active
10 Queue listening&waiting to join session
20 Listeners

Advantage:
More people can listen into sessions. 
Open Mic/Stage Pub Situation.

Once an active musicians leaves the next from wait queue can join.
Move to Pub Stage.

Listeners cannot be active. Pub Beer or Water drinking...

**Comment by ann0see** [comment]
I think the main issue here is that listening clients take up a server slot (technical limit) or do you think it’s a behavioural Problem?

Please note that social features have been rejected so far since they’re much more easily handled via websites/external tools (https://jamulus.live) —> KISS!
JSON RPC + Chat bots might in future give some integration?
Streaming (like jamcaster by @dtinth) will enable a lot more listeners – even without Jamulus. 

If it’s a technical problem I think it already works if you connect to a full server as soon as a slot gets free, the client gets audio.

**Comment by pljones** [comment]
Indeed, just set the expected behaviour in the server welcome message.

**Comment by reinhardwh** [comment]
@ann0see & @pljones 
Thanks for sharing your thoughts.
Will look into your suggestions. 😄

**Comment by gilgongo** [comment]
For the "listen not play" use case, you can also try something like this for example:

Server -> Client -> JACK ->  DarkIce -> ShoutCast

Not tried it myself though...

**Reply by rdica** [reply]
It works, I've used it as source for icecast.

**Reply by reinhardwh** [reply]
Hello @rdica,

do you have a more detailed description on how to set this up?

Jamulus > Client > Jack is ok

The rest is unknown to me

**Reply by rdica** [reply]
http://darkice.org
```
DarkIce is a live audio streamer. It records audio from an audio interface (e.g. sound card), encodes it and sends it to a streaming server.
DarkIce can record from:
OSS audio devices
ALSA audio devices
Solaris audio interface
Jack sources
uLaw audio input through a serial interface
CoreAudio (branch darkice-macosx)
```
Install darkice, create a configuration for the darkice client, start it, connect jamulus client output jack ports to darkice input ports. Darkice sends whatever is input to the stream server you configured.
A basic darkice.conf may be:

```
[general]
duration = 0
bufferSecs = 10

[input]
device = jack
jackClientName = darkice
sampleRate = 48000
bitsPerSample = 16
channel = 2

# icecast ogg
[icecast2-0]
bitrateMode = vbr
format = vorbis
bitrate = 256
quality = 1.0
server = servername_or_IP
port = port_num
password = h4ckm3
mountPoint = some_name.ogg
name = some stream name
description = some stream description
url = some stream url
genre = some_genre
public = yes_no
```

**Reply by reinhardwh** [reply]
Appreciated
Will test it and let you know if ... 🎸

**Reply by dtinth** [reply]
Also for the "listen not play" use case, I’ve written this guide some time ago, using `ffmpeg` on the command line instead of DarkIce.

https://dtinth.notion.site/Jamulus-Linux-Icecast-streaming-setup-fe47cded1ff2488cb9de9a0697e90a35

**Reply by reinhardwh** [reply]
Hello dtinth,

thanks for the tip!
